src/ImageSharp/Formats/Jpeg/JpegEncoderCore.cs

```diff
@@ -352,26 +349,18 @@ private void WriteXmpProfile(XmpProfile xmpProfile)
             }
 
             int dataLength = data.Length;
-            int offset = 0;
 
-            while (dataLength > 0)
+            // The XMP data cannot be split in multiple segments, therefore cap the data at 65KB.
```

Comment:
[It actually can be split](https://github.com/adobe/xmp-docs/blob/master/XMPSpecifications/XMPSpecificationPart3.pdf):
> If the serialized XMP packet becomes larger than the 64K limit, you can divide it into a main portion
(StandardXMP) and an extended portion (ExtendedXMP), and store it in multiple JPEG marker segment.

Reply:
Thanks for this info, I did not know that. My info came from https://dev.exiv2.org/projects/exiv2/wiki/The_Metadata_in_JPEG_files

Switching back to WIP.

Reply:
I think we can merge this PR as it is now to eliminate exceptions and provide split metadata implementation later.

Reply:
After reviewing this again, I think the current implementation in `main` is correct: It splits already the XMP data over multiple APP1 markers. My assumption that it should be capped at 65KB was wrong. I think I will close this PR then.

If anyone else wants to double check this (looking at you @br3aker :-), feel free to do so. A second opinion is always helpful.

Reply:
I think 65k is the cap for each segment. There's no limit to the actual size that I know of.